Simplified/AxisService/NYPLAxisService.swift

```diff
@@ -75,16 +129,194 @@ class NYPLAxisService: NSObject {
+  private func downloadAssetsFromPackage(forBook book: NYPLBook) {
+    dispatchGroup.wait()
+    guard let packageEndpoint = packageEndpoint else {
+      // No need to log error here since an error is already logged when
+      // pacakgeEndpoint generation returns nil
+      delegate?.failDownloadWithAlert(forBook: self.book)
+      return
+    }
+    
+    let packageURL = self.dedicatedWriteURL.appendingPathComponent(packageEndpoint)
+    let data = try? Data(contentsOf: packageURL)
+    guard let axisXML = NYPLAxisXML(data: data) else {
+      NYPLErrorLogger.logError(
+        withCode: .axisDRMFulfillmentFail,
+        summary: NYPLAxisService.failedDownloadingAssetsFromPackageSummary,
+        metadata: [
+          NYPLAxisService.reason: NYPLAxisXML.NYPLXMLGenerationFailure
+      ])
+      
+      return
+    }
+    
+    let hrefs = axisXML.findRecursivelyInAttributes(AxisHelper.hrefKey)
+    
+    for href in hrefs {
+      let endpath: String
+      if let pathPrefix = packagePathPrefix {
+        endpath = "\(pathPrefix)\(href)"
+      } else {
+        endpath = href
+      }
+      
+      let linkURL = baseURL
+        .appendingPathComponent(endpath)
+      
+      let writeURL = self.dedicatedWriteURL
+        .appendingPathComponent(endpath)
+      
+      dispatchGroup.enter()
+      downloadItem(from: linkURL, at: writeURL)
+    }
+  }
+  
+  // MARK: - Helper Methods
+  private func downloadItem(from url: URL, at writeURL: URL) {
+    
+    // TODO: OE-62: Axis code might need specific requirements for caching and
+    // using the shared network executor will result in whatever we choose to
+    // user there.
+    
+    NYPLNetworkExecutor.shared.GET(url) { (result) in
+      switch result {
+      case .success(let data, _):
+        do {
+          try NYPLAssetWriter.writeAsset(data, atURL: writeURL)
+          self.dispatchGroup.leave()
+        } catch {
+          
+          NYPLErrorLogger.logError(
+            error,
+            summary: NYPLAxisService.writeFailureSummary,
+            metadata: ["writeURL": writeURL.path,
+                       "itemURL": url.absoluteString])
+          
+          self.dispatchGroup.leave()
+          self.delegate?.failDownloadWithAlert(forBook: self.book)
+        }
+      case .failure(let error, _):
+        NYPLErrorLogger.logError(
+          error,
+          summary: NYPLAxisService.failedDownloadingItemSummary,
+          metadata: ["itemURL": url.absoluteString])
+        
+        self.dispatchGroup.leave()
+        self.delegate?.failDownloadWithAlert(forBook: self.book)
+      }
+    }
+  }
+  
+  private var packageEndpoint: String? {
+    if self.packageEndpointExtractor == nil {
+      let containerURL = dedicatedWriteURL
+        .appendingPathComponent(AxisHelper.containerFileName)
+      self.packageEndpointExtractor = NYPLAxisPackageEndpointProvider(
+        containerURL: containerURL,
+        fullPathKey: AxisHelper.fullPathKey)
+    }
```

Comment:
This needs a little fixing

Reply:
No need to have it as nil since we're relying on constant values to initiate packageEndpointExtractor